spec/lib/identity_config_spec.rb

```diff
@@ -73,4 +77,53 @@
       expect(Identity::Hostdata.config_builder.unused_keys).to be_empty
     end
   end
+
+  describe 'redundant configuration' do
+    it 'does not use the default value in development' do
+      aggregate_failures do
+        default_yaml_config['development'].each do |key, value|
+          next unless default_yaml_config.key?(key)
+          expect(value).to_not eq(default_yaml_config[key]),
+                               "#{key} in development uses the default value"
+        end
+      end
```

Comment:
since this is repeated 3x, maybe could extract into a helper?
```ruby
def check_for_default(env_name)
  aggregate_failures do
    default_yaml_config[env_name].each do |key, value|
      next unless default_yaml_config.key?(key)
      expect(value).to_not eq(default_yaml_config[key]), "#{key} in #{env_name} uses the default value"
    end
  end
end
```

Reply:
Good call, added in 72f5638920